crates/jstz_core/src/native.rs

```diff
@@ -424,14 +417,13 @@ fn raw_constructor<T: NativeClass>(
         .transpose()?
         .flatten();
 
-    let native_object = T::constructor(
-        &JsNativeObject::from_value_unchecked(this.clone()),
-        args,
-        context,
-    )?;
+    let native_object = T::data_constructor(target, args, context)?;
+
     let object =
```

Comment:
I find the naming a bit confusing here. What's the differnce between instance and native object? 

the variable `object` is of type `JsNativeObject` which is a type safe version of `native_object`?

Reply:
`native_object` here refers to the native Rust value

Reply:
`object` is a JavaScript object (wrapped using the `JsNativeObject` struct -- which gives some more type safety)